Packs/DomainToolsIrisDetect/Integrations/DomainToolsIrisDetect/DomainToolsIrisDetect.py

```diff
@@ -0,0 +1,1347 @@
+            last_run (str): A timestamp string indicating the last run.
+            enable_incidents (bool): Specifies whether to create an incident or not. Default is True.
+
+        Returns:
+            List[Dict[str, Any]]: A list containing the incident object if one was created,
+            otherwise an empty list.
+        """
+
+        indicators = [self.create_indicator_from_detect_domain(item, term) for item in domains_list]
+        if not indicators:
+            return []
+
+        for batched in batch(indicators, batch_size=BATCH_SIZE):
+            demisto.createIndicators(batched)
+        demisto.info(f"Added {len(indicators)} indicators to demisto")
+
+        if enable_incidents:
+            last_run_dt_without_ms = datetime.strptime(get_last_run(last_run), DATE_FORMAT).replace(
+                microsecond=0) if get_last_run(last_run) else None
+            first_run_dt_without_ms = (datetime.now() - timedelta(days=validate_first_fetch(self.first_fetch))).replace(
+                microsecond=0)
+            incident = {
+                "name": f"{incident_name} "
+                        f"{last_run_dt_without_ms or first_run_dt_without_ms}",
+                "details": json.dumps(domains_list),
+                "rawJSON": json.dumps({"incidents": domains_list}),
+                "type": INCIDENT_TYPE[incident_name],
+            }
+            return [incident]
+
+        return []
+
+    def fetch_dt_domains_from_api(self, end_point: str, last_run: str) -> Tuple[List[Dict], str]:
+        """
+        Makes an API call to the Domain Tools API endpoint and retrieves domain data based on the provided
+        parameters.
+
+        Args:
+            end_point (str): The API endpoint to call.
+            last_run (str): The timestamp of the last successful API call.
+
+        Returns:
+            Tuple[List[Dict], str]: A tuple containing a list of watchlist domains and a timestamp of the current
+            API call.
+
+        """
+
+        last_run_value = get_last_run(last_run)
+        if last_run_value:
+            params = DOMAINTOOLS_PARAMS | {
+                DT_TIMESTAMP_DICT[last_run]: last_run_value,
+                "include_domain_data": INCLUDE_DOMAIN_DATA_VALUE if self.include_domain_data else 0,
+            }
+            demisto.info(f'Found last run, fetching domains from {last_run_value}')
+        else:
+            days_back = validate_first_fetch(self.first_fetch)
+            params = DOMAINTOOLS_PARAMS | {
+                DT_TIMESTAMP_DICT[last_run]: datetime.now() - timedelta(days=days_back),
+                "include_domain_data": INCLUDE_DOMAIN_DATA_VALUE if self.include_domain_data else 0,
+            }
+            demisto.info(f'First run, fetching domains from last {days_back} days')
+
+        if self.risk_score_ranges:
+            params["risk_score_ranges[]"] = self.risk_score_ranges
+
+        results: List = []
+        while True:
```

Comment:
Please use **self.fetch_limit** instead of **while True**

Reply:
As we have explained  we are saving only three incidents, this loop is for domains in the incidents